### PR TITLE
Prevent modal top from shrinking

### LINKED ISSUE #7166: Collection share dialog shrinking
Collection share dialog shrinks vertically to remain center aligned as search query is typed. It shouldn't shrink from top.

https://github.com/outline/outline/assets/15694566/afda0f93-48f6-4e4b-a98c-d1985db5598f

### PR DESCRIPTION
**Pull request by apoorv-mishra**
Closes #7166

### CODE REVIEW

File: app/components/Modal.tsx [hunk 1]
```diff
@@ -254,7 +254,7 @@ const Header = styled(Flex)`
 const Small = styled.div`
   animation: ${fadeAndScaleIn} 250ms ease;
 
-  margin: auto auto;
+  margin: 25vh auto auto auto;
```
Comment (apoorv-mishra) [inline app/components/Modal.tsx:257]: Note that this affects dialogs across the board.

Comment (tommoor) [reply app/components/Modal.tsx:257]: I believe we need to define a max height too, maybe 70vh?

image

Comment (apoorv-mishra) [reply app/components/Modal.tsx:257]: Just a note on this - setting max height leads us to manage overflow which provokes to manage scroll 😅 

Looked into all this yesterday, almost there...

Comment (apoorv-mishra) [reply app/components/Modal.tsx:257]: 🤞 


https://github.com/outline/outline/assets/15694566/68b03146-af78-4dc5-a3a6-992146a28710

File: app/components/List/Item.tsx [hunk 2]
```diff
@@ -47,22 +48,33 @@ const ListItem = (
...
-  useFocusEffect(focused, itemRef as React.RefObject<HTMLAnchorElement>);
+  useFocusEffect(focused, itemRef);
+
+  const handleFocus = React.useCallback(() => {
+    if (itemRef.current) {
+      scrollIntoView(itemRef.current, {
+        scrollMode: "if-needed",
+        behavior: "auto",
+        block: "center",
+        boundary: window.document.body,
```
Comment (tommoor) [inline app/components/List/Item.tsx:74]: Isn't this boundary the default?

File: app/hooks/useMaxHeight.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import * as React from "react";
+import useMobile from "./useMobile";
+import useWindowSize from "./useWindowSize";
+
+const useMaxHeight = ({
+  elementRef,
+  maxViewportHeight = 90,
+  margin = 8,
```
Comment (tommoor) [inline app/hooks/useMaxHeight.ts:8]:
```suggestion
  margin = 16,
```

File: app/hooks/useMaxHeight.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import * as React from "react";
+import useMobile from "./useMobile";
+import useWindowSize from "./useWindowSize";
+
+const useMaxHeight = ({
+  elementRef,
+  maxViewportHeight = 90,
```
Comment (tommoor) [inline app/hooks/useMaxHeight.ts:7]: I don't love this naming, might be better to pass as a fraction or… `maxViewportPercentage` for clarity

File: app/components/List/Item.tsx (hunk 2 above)
Comment (apoorv-mishra) [reply app/components/List/Item.tsx:74]: Doesn't look like. The scroll isn't prevented upon keyboard navigation if `boundary` is absent. This fixes the problem.

**Review by tommoor — APPROVED** [review]

**Pull request merged**